### PR TITLE
Updated dockerfile and helm values to support amd64 and arm64

### LINKED ISSUE #462: Remote values from http is not working
### Current Behavior
```
Failed to load target state: failed to generate manifest for source 2 of 2: rpc error: code = Unknown desc = `helm template . --name-template argocd --namespace argocd --kube-version 1.30 --values secrets://https://raw.githubusercontent.com/ajaykumar4/home-lab-argocd/main/kubernetes/apps/argocd/argocd/secrets.sops.yaml <api versions removed> --include-crds` failed exit status 1: /custom-tools/helm-plugins/helm-secrets/scripts/run.sh: 7: /custom-tools/curl: Exec format error [helm-secrets] Error while download url https://raw.githubusercontent.com/ajaykumar4/home-lab-argocd/main/kubernetes/apps/argocd/argocd/secrets.sops.yaml [helm-secrets] File does not exist: https://raw.githubusercontent.com/ajaykumar4/home-lab-argocd/main/kubernetes/apps/argocd/argocd/secrets.sops.yaml Error: failed parsing --set-file data: plugin "scripts/run.sh downloader" exited with error [helm-secrets] helm template command errored on value 'secrets://https://raw.githubusercontent.com/ajaykumar4/home-lab-argocd/main/kubernetes/apps/argocd/argocd/secrets.sops.yaml' [helm-secrets] File does not exist: secrets://https://raw.githubusercontent.com/ajaykumar4/home-lab-argocd/main/kubernetes/apps/argocd/argocd/secrets.sops.yaml Error: plugin "secrets" exited with error
```
### Expected Behavior

it should have pulled the values file from github 

### Steps To Reproduce

```markdown
application.yaml
---
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: argocd
  namespace: argocd
  annotations:
    argocd.argoproj.io/sync-wave: '4'
spec:
  project: home-kubernetes
  sources:
    - repoURL: https://github.com/ajaykumar4/home-lab-argocd.git
      path: kubernetes/apps/argocd/argocd
      targetRevision: main
      ref: repo
    - repoURL: ghcr.io/argoproj/argo-helm
      chart: argo-cd
      targetRevision: 7.3.4
      helm:
        releaseName: argocd
        valueFiles:
          - secrets://https://raw.githubusercontent.com/ajaykumar4/home-lab-argocd/main/kubernetes/apps/argocd/argocd/secrets.sops.yaml
  destination:
    name: in-cluster
    namespace: argocd
  syncPolicy:
    automated:
      allowEmpty: true
      prune: true
      selfHeal: true
    syncOptions:
      - CreateNamespace=true
      - ServerSideApply=true
      - RespectIgnoreDifferences=true
  ignoreDifferences:
  - group: argoproj.io
    kind: Application
    jsonPointers:
    - /status
    - /metadata/annotations
```

### Environment

- Helm Version: v3.14.4
- Helm Secrets Version: 4.6.0
- ArgoCD Version: v2.11.4+e1284e1
- OS:
- Shell:

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ajaykumar4**
fixes #462

### CODE REVIEW

**Comment by jkroepke** [comment]
Could you explain, why you are manually download curl instead using the os one?

**Comment by ajaykumar4** [comment]
To keep init container and dockerfile straps as same as possible

File: docs/ArgoCD Integration.md [hunk 1]
```diff
@@ -166,21 +167,31 @@ ENV HELM_SECRETS_BACKEND="vals" \
...
+    GO_ARCH=$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/') && \
+    wget -qO "/gitops-tools/kubectl"   "https://dl.k8s.io/release/${KUBECTL_VERSION}/bin/linux/${GO_ARCH}/kubectl" && \
+    true
 
 # sops backend installation (optional)
-RUN curl -fsSL https://github.com/getsops/sops/releases/download/v${SOPS_VERSION}/sops-v${SOPS_VERSION}.linux.amd64 \
-    -o /usr/local/bin/sops && chmod +x /usr/local/bin/sops
+RUN \
+    GO_ARCH=$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\(arm\)\(64\)\?.*/\1\2/' -e 's/aarch64$/arm64/') && \
+    wget -qO "/gitops-tools/sops"      "https://github.com/mozilla/sops/releases/download/${SOPS_VERSION}/sops-${SOPS_VERSION}.linux.${GO_ARCH}" && \
```
Comment (jkroepke) [inline docs/ArgoCD Integration.md:187]: In general, I like the idea, to move sops from /usr/local/bin/sops to /gitops-tools/sops, but than helm-secrets needs to know this.

Example from helm values below

```
    - name: HELM_SECRETS_CURL_PATH
      value: /custom-tools/curl
    - name: HELM_SECRETS_SOPS_PATH
      value: /custom-tools/sops
    - name: HELM_SECRETS_VALS_PATH
      value: /custom-tools/vals
    - name: HELM_SECRETS_KUBECTL_PATH
      value: /custom-tools/kubectl
```

**Review by jkroepke — CHANGES_REQUESTED** [review]
the binary path location must be defined

File: docs/ArgoCD Integration.md (hunk 1 above)
Comment (ajaykumar4) [reply docs/ArgoCD Integration.md:187]: Will update it

File: docs/ArgoCD Integration.md [hunk 2]
```diff
@@ -166,27 +167,49 @@ ENV HELM_SECRETS_BACKEND="vals" \
...
 
-RUN ln -sf "$(helm env HELM_PLUGINS)/helm-secrets/scripts/wrapper/helm.sh" /usr/local/sbin/helm
+RUN chmod +x /gitops-tools/* && ln -sf /gitops-tools/helm-plugins/helm-secrets/scripts/wrapper/helm.sh /usr/local/sbin/helm
+
+ENV PATH="$PATH:/gitops-tools"
+ENV HELM_PLUGINS=/gitops-tools/helm-plugins/
+ENV HELM_SECRETS_CURL_PATH=/gitops-tools/curl
+ENV HELM_SECRETS_SOPS_PATH=/gitops-tools/sops
+ENV HELM_SECRETS_VALS_PATH=/gitops-tools/vals
+ENV HELM_SECRETS_KUBECTL_PATH=/gitops-tools/kubectl
```
Comment (jkroepke) [inline docs/ArgoCD Integration.md:209]: could you please move all env together to line 156? not worth to have one extra command for each env

**Review by jkroepke — APPROVED** [review]

**Comment by jkroepke** [comment]
Thanks

**Pull request merged**